src/app/shared/components/gon-list/gon-list.component.ts

```diff
@@ -1,58 +1,39 @@
-import { Component, OnInit, Input, OnChanges } from "@angular/core";
+import { Component, OnInit, Input, OnChanges, Output, EventEmitter } from "@angular/core";
 import { GonListData } from "./../../../difs/gon-list-data";
 @Component({
   selector: "app-gon-list",
   templateUrl: "./gon-list.component.html",
   styleUrls: ["./gon-list.component.scss"],
 })
-export class GonListComponent implements OnInit, OnChanges {
+export class GonListComponent implements  OnChanges {
   constructor() {}
   @Input() listData: GonListData[];
+  @Output() clickData = new EventEmitter<Object>();
 
   clickedObj = {};
```

Comment:
如果要做紀錄當前正點擊哪個項目，用currentClickIndex的方式去記可能會更好些，-1代表都沒選任何項目。
```suggestion
  currentClickIndex :number = -1;
```